File: runtimes/dotnet-7.0/src/RuntimeResponse.cs
using System;
using System.Collections.Generic;
using System.Text.Json;

namespace DotNetRuntime {
    public class RuntimeResponse
    {
        public RuntimeOutput Send(string body, int statusCode = 200, Dictionary<string, string>? headers = null)
        {
            if(headers == null) {
                headers = new Dictionary<string,string>();
            }

            if (!headers.ContainsKey("content-type"))
            {
                headers.Add("content-type", "text/plain");
            }

            return new RuntimeOutput(
                body,
                statusCode,
                headers);
        }

        public RuntimeOutput Json(Dictionary<string, object?> json, int statusCode = 200, Dictionary<string, string>? headers = null)
        {
            if(headers == null) {
                headers = new Dictionary<string,string>();
            }

            if (!headers.ContainsKey("content-type"))
            {
                headers.Add("content-type", "application/json");
            }

            return Send(JsonSerializer.Serialize(json), statusCode, headers);
        }

        public RuntimeOutput Empty()
        {
            return new RuntimeOutput("", 204, new Dictionary<string,string>());
        }

        public RuntimeOutput Redirect(String url, int statusCode = 301, Dictionary<string, string>? headers = null)
        {
            if(headers == null) {
                headers = new Dictionary<string,string>();
            }

            headers.Add("location", url);

            return Send("", statusCode, headers);
        }
    }
}

File: runtimes/swift-5.5/Sources/Runtime/RuntimeResponse.swift
import Foundation

class RuntimeResponse {

    func send(
        _ body: String,
        statusCode: Int = 200,
        headers: [String: String] = [:]
    ) -> RuntimeOutput {
        return RuntimeOutput(
            body: body,
            statusCode: statusCode,
            headers: headers
        )
    }

    func json(
        _ json: [String: Any?],
        statusCode: Int = 200,
        headers: [String: String] = [:]
    ) throws -> RuntimeOutput {
        var outputHeaders = headers
        
        if outputHeaders["content-type"] == nil {
            outputHeaders["content-type"] = "application/json"
        }
        
        let data = try JSONSerialization.data(
          withJSONObject: json,
          options: .prettyPrinted
        )

        let text = String(
            data: data,
            encoding: .utf8
        )

        return RuntimeOutput(
            body: text!,
            statusCode: statusCode,
            headers: outputHeaders
        )
    }

    func empty() -> RuntimeOutput {
        return RuntimeOutput(
            body: "",
            statusCode: 204,
            headers: [:]
        )
    }

    func redirect(
        _ url: String,
        statusCode: Int = 301,
        headers: [String: String] = [:]
    ) -> RuntimeOutput {
        var outputHeaders = headers

        outputHeaders["location"] = url

        return RuntimeOutput(
            body: "",
            statusCode: statusCode,
            headers: outputHeaders
        )
    }
}
